Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.h

```diff
@@ -10,6 +10,30 @@
 
 namespace Gfx {
 
+namespace JPEG2000 {
+
+struct TagTreeNode;
+class TagTree {
+public:
+    TagTree(TagTree&&);
+    ~TagTree();
+
+    TagTree(TagTree const&) = delete;
+    TagTree& operator=(TagTree const&) = delete;
+
+    static ErrorOr<TagTree> create(u32 x_count, u32 y_count);
+
+    ErrorOr<u32> read_value(u32 x, u32 y, Function<ErrorOr<bool>()> const& read_bit, Optional<u32> stop_at = {}) const;
+
+private:
+    TagTree();
+
+    Vector<TagTreeNode> nodes;
+    TagTreeNode* root { nullptr };
```

Comment:
Should probably be a reference, or omitted alltogether (since it appears to always be `nodes[0]`).

Reply:
Sure, made it a reference. (Currently all knowledge of the layout of `nodes` is in `TagTree::create()`, which seems nice to me. So I prefer keeping it.)